src/lib/reasoners/arith.ml

```diff
@@ -813,23 +813,27 @@ module Shostak
         cpt := Q.add Q.one (max_constant distincts !cpt);
         Some (term_of_cst (Q.to_string !cpt), true)
 
-
-
-  let pprint_const_for_model =
-    let pprint_positive_const c =
-      let num = Q.num c in
-      let den = Q.den c in
-      if Z.is_one den then Z.to_string num
-      else Format.sprintf "(/ %s %s)" (Z.to_string num) (Z.to_string den)
-    in
-    fun r ->
-      match P.is_const (embed r) with
-      | None -> assert false
-      | Some c ->
-        let sg = Q.sign c in
-        if sg = 0 then "0"
-        else if sg > 0 then pprint_positive_const c
-        else Format.sprintf "(- %s)" (pprint_positive_const (Q.abs c))
+  let pp_positive_real ppf q =
+    if Z.equal (Q.den q) Z.one then
+      Fmt.pf ppf "%a.0" Z.pp_print (Q.num q)
+    else
+      Fmt.pf ppf "(/ %a.0 %a.0)" Z.pp_print (Q.num q) Z.pp_print (Q.den q)
```

Comment:
Regardless of how we deal with the integer case, fractions should be `(/ 1 4)`, not `(/ 1.0 4.0)`.

Reply:
Dolmen has separate grammars for scripts and responses. It accepts terms such as `(/ 1 4)` in models (it would be surprising if it didn't since that is both what the spec says and the way Z3 and CVC5 display rationals).

Reply:
Seriously... it's ... bang

Reply:
Technically, dolmen has the same grammar/syntax for both scripts and response, and the typechecking is mostly the same (the biggest difference would be that most of the linearity warnings are silenced during models, since it should not matter).

Reply:
Yes, I meant that things happen differently in both cases. Hm, but it is still true in my tests that `(/ 1 4)` is accepted by Dolmen in models and actually it seems to also accept it in scripts on my machine?